services/order.py

```diff
@@ -0,0 +1,33 @@
+from django.db import transaction
+from django.contrib.auth import get_user_model
+from db.models import Order, MovieSession, Ticket
+
+
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None) -> Order:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        order = Order.objects.create(user=user)
+        if date is not None:
+            order.created_at = date
+        for ticket in tickets:
+            row = ticket["row"]
+            seat = ticket["seat"]
+            movie_session_id = ticket["movie_session"]
+            movie_session = MovieSession.objects.get(id=movie_session_id)
+            Ticket.objects.create(
+                movie_session=movie_session,
+                order=order,
+                row=row,
+                seat=seat
+            )
+    order.save()
+    return order

```

Comment:
use transaction.atomic decorator. The decorator is a more concise and cleaner way to ensure that the function executes within a transaction.